src/rpc/mining.cpp

```diff
@@ -14,19 +14,73 @@
 #include "net.h"
 #include "pow.h"
 #include "rpc/server.h"
-#include "util.h"
 #include "validationinterface.h"
 #ifdef ENABLE_WALLET
 #include "wallet/db.h"
 #include "wallet/wallet.h"
 #endif
-#include "warnings.h"
 
 #include <stdint.h>
 
 #include <univalue.h>
 
 #ifdef ENABLE_WALLET
+UniValue generateBlocks(const Consensus::Params& consensus,
```

Comment:
Since we are extracting this logic into its own helper function, would be good to pass the wallet from the caller (as per https://github.com/PIVX-Project/PIVX/pull/2337/commits/5b26648b38e075bda57ca6b21a77df64d1d9b839).

But if you don't want to do it, and this gets merged before, I can do it inside #2337.